packages/design-system/src/components/InlineEditing/Primitives/InlineEditingPrimitive.tsx

```diff
@@ -84,6 +86,14 @@ const InlineEditingPrimitive = forwardRef(
 			}
 		}, [hasError]);
 
+		useEffect(() => {
+			// Update state's value when default value is updated
+			if (value !== defaultValue) setValue(defaultValue);
```

Comment:
nop the idea of a defaultValue is to uncontrolled.
If you want to control the value we should use a new value props and use this one here.